### PR TITLE
Formatter

### PR DESCRIPTION
**Pull request by ldlsegovia**
Lo explico en detalle en el README pero en resumen:
Hice un formatter para simple cov que genera un archiv json que lo lee un plugin de vscode y permite resaltar líneas de código que no hemos cubierto con nuestros tests.

covered
not-covered
partial-cov

### CODE REVIEW

File: lib/simplecov_linter_formatter/formatters/source_file_formatter.rb [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    end
+
+    private
+
+    def build_message(line)
+      [
+        @source_file.filename,
+        line.line_number,
+        FILE_COLUMN,
+        "#{line.status}-#{lines_count}",
```
Comment (monkeyci) [inline lib/simplecov_linter_formatter/formatters/source_file_formatter.rb:22]: **[rubocop]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>[Correctable] Style/TrailingCommaInArrayLiteral: Avoid comma after the last item of an array.

File: lib/simplecov_linter_formatter/formatters/text_lines_formatter.rb [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+      }
+    end
+
+    private
+
+    def group_lines_by_file
+      result = {}
+
+      @lines.each do |line|
+        file_name, line_number, column, status_lines_count = line.split(":")
```
Comment (monkeyci) [inline lib/simplecov_linter_formatter/formatters/text_lines_formatter.rb:22]: ⚠️ **[rubocop]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Lint/UselessAssignment: Useless assignment to variable - column. Use _ or _column as a variable name to indicate that it won't be used.

File: spec/lib/formatters/text_lines_formatter_spec.rb [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+describe SimpleCovLinterFormatter::TextLinesFormatter do
+  let(:command_name) { "RSpec" }
+  let(:lines) do
+    [
+      "file1.rb:1:1:missed-2",
+      "file1.rb:2:1:never-2",
+      "file2.rb:1:1:missed-4",
+      "file2.rb:2:1:covered-4",
+      "file2.rb:4:1:missed-4",
```
Comment (monkeyci) [inline spec/lib/formatters/text_lines_formatter_spec.rb:11]: **[rubocop]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>[Correctable] Style/TrailingCommaInArrayLiteral: Avoid comma after the last item of an array.

File: spec/lib/text_lines_filter_spec.rb [hunk 4]
```diff
@@ -0,0 +1,44 @@
+require "spec_helper"
+
+describe SimpleCovLinterFormatter::TextLinesFilter do
+  let(:lines) do
+    [
```
Comment (monkeyci) [inline spec/lib/text_lines_filter_spec.rb:5]: **[rubocop]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>[Correctable] Style/WordArray: Use %w or %W for an array of words.

File: spec/lib/text_lines_filter_spec.rb [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+      "line1",
+      "line2",
```
Comment (monkeyci) [inline spec/lib/text_lines_filter_spec.rb:7]: **[rubocop]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>[Correctable] Style/TrailingCommaInArrayLiteral: Avoid comma after the last item of an array.

File: spec/lib/text_lines_filter_spec.rb [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+    ]
+  end
+
+  let(:filtered_lines) { "filtered_line1\nfiltered_line2" }
+
+  let(:expected_result) do
+    [
```
Comment (monkeyci) [inline spec/lib/text_lines_filter_spec.rb:14]: **[rubocop]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>[Correctable] Style/WordArray: Use %w or %W for an array of words.

File: spec/lib/text_lines_filter_spec.rb [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+      "filtered_line1",
+      "filtered_line2",
```
Comment (monkeyci) [inline spec/lib/text_lines_filter_spec.rb:16]: **[rubocop]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>[Correctable] Style/TrailingCommaInArrayLiteral: Avoid comma after the last item of an array.

**Review by iaacosta — APPROVED** [review]
Que la raja la gema! 💯

**Comment by ldlsegovia** [comment]
> Que la raja la gema! 💯

Ojalá la usemos y de resultados :)

**Pull request merged**